go/database/mpt/tool/export_test.go

```diff
@@ -0,0 +1,217 @@
+//
+// Change Date: 2028-4-16
+//
+// On the date above, in accordance with the Business Source License, use of
+// this software will be governed by the GNU Lesser General Public License v3.
+
+package main
+
+import (
+	"context"
+	"errors"
+	"io"
+	"path/filepath"
+	"syscall"
+	"testing"
+	"time"
+
+	"github.com/Fantom-foundation/Carmen/go/common"
+	"github.com/Fantom-foundation/Carmen/go/database/mpt"
+	cio "github.com/Fantom-foundation/Carmen/go/database/mpt/io"
+)
+
+func TestExport_CanBeInterrupted(t *testing.T) {
+	type testFuncs struct {
+		export   func(context.Context, string, io.Writer) error
+		createDB func(t *testing.T, dir string)
+		check    func(t *testing.T, dir string) error
+	}
+	tests := map[string]testFuncs{
+		"archive": {
+			export:   cio.ExportArchive,
+			createDB: createArchiveTrie,
+			check:    checkExportedArchive,
+		},
+		"live": {
+			export:   cio.Export,
+			createDB: createLiveTrie,
+			check:    checkExportedLiveDB,
+		},
+	}
+
+	for name, tf := range tests {
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+			dir := filepath.Join(t.TempDir(), name)
+			tf.createDB(t, dir)
+			ctx, cancel := context.WithCancel(context.Background())
+			catchInterrupt(ctx, cancel, time.Now())
+
+			success := make(chan any, 1)
+			writer := newMockWriter(t)
+
+			// first find number of writes
+			if err := tf.export(ctx, dir, writer); err != nil {
+				t.Errorf("unexpected error: %v", err)
+				return
+			}
+
+			// save max count and reset number of writes
+			maxCount := writer.numOfWrites
+			writer.numOfWrites = 0
+
+			go func() {
+				if err := tf.export(ctx, dir, writer); err != nil {
+					t.Errorf("unexpected error: %v", err)
+					return
+				}
+				close(success)
+			}()
+
+			time.Sleep(200 * time.Millisecond)
+			err := syscall.Kill(syscall.Getpid(), syscall.SIGINT)
+			if err != nil {
+				t.Error("failed to create a SIGINT signal")
+				return
+			}
+			select {
+			case <-success:
+				if maxCount > writer.numOfWrites {
+					break
+				}
+				t.Fatal("export was not interrupted")
+			case <-time.After(10 * time.Second):
+				t.Error("export was not interrupted")
+			}
+
+			err = tf.check(t, dir)
+			if err != nil {
+				t.Fatalf("db is corrupted after export interrupt: %v", err)
+			}
+		})
+	}
+
+}
+
+func checkExportedLiveDB(t *testing.T, dir string) error {
+	t.Helper()
+	return checkLiveDB(dir, cio.MptInfo{Config: mpt.S5LiveConfig})
+}
+func checkExportedArchive(t *testing.T, dir string) error {
+	t.Helper()
+	return checkLiveDB(dir, cio.MptInfo{Config: mpt.S5ArchiveConfig})
+}
+
+type mockWriter struct {
+	numOfWrites int
+}
+
+func newMockWriter(t *testing.T) *mockWriter {
+	t.Helper()
+	return &mockWriter{}
+}
+
+func (m *mockWriter) Write(_ []byte) (n int, err error) {
+	m.numOfWrites++
+	// slow down writting to be able to interrupt the export
+	time.Sleep(100 * time.Millisecond)
+	return 0, nil
+}
+
+func createLiveTrie(t *testing.T, dir string) {
```

Comment:
if the test is in the `io` package, there is already functionality to create test data